sdk/nanotdf.go

```diff
@@ -652,6 +649,33 @@ func NewNanoTDFHeaderFromReader(reader io.Reader) (NanoTDFHeader, uint32, error)
 	return header, size, nil
 }
 
+func nonZeroRandomPaddedIV() ([]byte, error) {
+	const (
+		loopCountLimit = 10
+	)
+	loopCount := 1
+	for {
+		ivPadded := make([]byte, 0, ocrypto.GcmStandardNonceSize)
+		noncePadding := make([]byte, kIvPadding)
+		ivPadded = append(ivPadded, noncePadding...)
+		iv, err := ocrypto.RandomBytes(kNanoTDFIvSize)
+		if err != nil {
+			return nil, fmt.Errorf("ocrypto.RandomBytes failed:%w", err)
+		}
+		ivPadded = append(ivPadded, iv...)
+		for _, b := range ivPadded {
+			if b != 0 {
+				return ivPadded, nil
+			}
+		}
+		// all zero IV, this is extremely rare so should be able to be addressed in the next loop
+		if loopCount >= loopCountLimit { // crazy, there must be an issue with the constants
+			return nil, errors.New("nonZeroPaddedIV loop exceeded limit")
```

Comment:
Should we expose this error or a similar one as fatal? This way developers can test for `error.Is(sdk.ErrNanoFatalZeroPaddedIv)`

Reply:
It's hard to imagine a condition where this error case is hit.  That said, I put the loop limit in because a tight loop would likely be harder to understand if a bug does develop here.

Personally I don't think it's worth elevating, but let me know your preference.

Reply:
That's fair.